diff --git a/src/nix/modules/service/docker-compose-service.nix b/src/nix/modules/service/docker-compose-service.nix
index eb48ae1..70cfabb 100644
--- a/src/nix/modules/service/docker-compose-service.nix
+++ b/src/nix/modules/service/docker-compose-service.nix
@@ -246,6 +246,21 @@ in
         ${serviceRef "user"}
       '';
     };
+    service.cpuset = mkOption {
+        type = nullOr str;
+        default = null;
+        description = serviceRef "cpuset";
+    };
+    service.cpus = mkOption {
+        type = nullOr str;
+        default = null;
+        description = serviceRef "cpus";
+    };
+    service.mem_limit = mkOption {
+        type = nullOr str;
+        default = null;
+        description = serviceRef "mem_limit";
+    };
     service.ports = mkOption {
       type = listOf types.unspecified;
       default = [];
@@ -508,6 +523,12 @@ in
     inherit (config.service) working_dir;
   } // lib.optionalAttrs (config.service.user != null) {
     inherit (config.service) user;
+  } // lib.optionalAttrs (config.service.cpuset != null) {
+    inherit (config.service) cpuset;
+  } // lib.optionalAttrs (config.service.cpus != null) {
+    inherit (config.service) cpus;
+  } // lib.optionalAttrs (config.service.mem_limit != null) {
+    inherit (config.service) mem_limit;
   } // lib.optionalAttrs (config.service.blkio_config != null) {
     blkio_config = config.service.blkio_config._out;
   };
